pkg/prometheus/operator.go

```diff
@@ -229,6 +229,12 @@ func (sr *StatusReporter) Process(ctx context.Context, p monitoringv1.Prometheus
 		}
 	}
 
+	// Indicate that no scrape resource selectors are specified. Do not take precedence over an unhealthy condition.
+	reconciledCondition := sr.Reconciliations.GetReconciledCondition(key, p.GetObjectMeta().GetGeneration())
+	if AllSelectorsNil(p) && reconciledCondition.Status == monitoringv1.ConditionTrue {
+		reconciledCondition.Reason = "NoScrapeResourceSelected"
+		reconciledCondition.Message = "The operator selected no ServiceMonitor, PodMonitor, Probe and ScrapeConfig resources because all resource selectors are null."
```

Comment:
Additionally we can check `additionalScrapeConfigs` and tweak the message depending on the value being nil or not.  
```suggestion
		reconciledCondition.Message = "The resource selects no ServiceMonitor, PodMonitor, Probe or ScrapeConfig resources. Scraped targets are configured via additionalScrapeConfigs."
		reconciledCondition.Message = "The resource selects no ServiceMonitor, PodMonitor, Probe or ScrapeConfig resources and additionalScrapeConfigs isn't defined either."
```